compactor/src/compact.rs

```diff
@@ -356,28 +366,132 @@ impl Compactor {
+            for ts in tombstones {
+                let tss = table_tombstones.entry(ts.table_id).or_insert_with(Vec::new);
+                tss.push(ts);
+            }
+
+            for (table_id, tombstones) in table_tombstones {
+                let mut parquet_file_with_tombstones: BTreeMap<
+                    ParquetFileId,
+                    ParquetFileWithTombstone,
+                > = BTreeMap::new();
+                for ts in tombstones {
+                    // Read parquet files that are not soft deleted, not tombstone-processed, same table and
+                    // overlaps with this tombstone
+                    // TODO - QUESTIONS:
+                    //    1. Should we only read level-1 files only (right now, we read all files that overlap with the tombstone)?
+                    //    2. Need to handle the case that one or many of these files are also included in other background cycle
+                    //       that compact hot partitions.
+                    let parquet_files = repos
+                        .parquet_files()
+                        .list_by_tombstone(ts.clone())
+                        .await
+                        .context(ListParquetFilesByTombstoneSnafu)?;
+
+                    // Attach this tombstone to all parquet files and add it to the parquet_file_with_tombstones if not yet
+                    for pf in parquet_files {
+                        let file = parquet_file_with_tombstones
+                            .entry(pf.id)
+                            .or_insert_with(|| ParquetFileWithTombstone {
+                                data: Arc::new(pf),
+                                tombstones: vec![],
+                            });
+                        file.add_tombstones(vec![ts.clone()]);
+                    }
+                }
+
+                // Group parquet_file_with_tombstones into partition and add it into the partitions
+                let mut table_partittions: BTreeMap<
+                    PartitionId,
+                    Vec<Arc<ParquetFileWithTombstone>>,
+                > = BTreeMap::new();
+                for (_, pf) in parquet_file_with_tombstones {
+                    let list = table_partittions
+                        .entry(pf.data.partition_id)
+                        .or_insert_with(Vec::new);
+                    list.push(Arc::new(pf));
+                }
+
+                let ps = table_partittions
+                    .into_iter()
+                    .map(
+                        |(partition_id, files_with_tombstones)| PartitionWithParquetFiles {
+                            sequencer_id: *sequencer_id,
+                            table_id,
+                            partition_id,
+                            files_with_tombstones,
+                        },
+                    )
+                    .collect::<Vec<_>>();
+
+                partitions.extend(ps);
```

Comment:
`Vec` implements [the `Extend` trait](https://doc.rust-lang.org/core/iter/trait.Extend.html), so it can be extended with an iterator, and we can save some time by not collecting into an intermediate vector here:

```suggestion
                    );

                partitions.extend(ps);
```